app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -26,112 +26,145 @@ class EventRepositoryImpl @Inject constructor(
     private val ioDispatcher: CoroutineDispatcher
 ) : EventRepository {
 
-    override suspend fun createEvent(event: AgendaItem.Event) {
-        db.getEventDao().upsertEvent(event.toEventEntity())
-        val result = getAuthResult {
-            api.createEvent(
-                eventData = MultipartBody.Part
-                    .createFormData(
-                        name = "create_event_request",
-                        value = Json.encodeToString(event.toCreateEventRequest())
-                    ),
-                photoData = event.photos.filterIsInstance<EventPhoto.Local>().mapIndexed {
-                    index, eventPhoto ->
-                    val uriByteConverter = UriByteConverter(
-                        appContext = appContext,
-                        ioDispatcher = ioDispatcher
-                    )
-                    val imageByte = uriByteConverter.uriToByteArray(
-                        uri = Uri.parse(eventPhoto.uri),
-                    )
-
-                    val compressedImageByte = BitmapCompressor().compressByteArray(
-                        byteArray = imageByte,
-                        targetSize = 1000000
-                    )
-                    MultipartBody.Part
+    override suspend fun upsertEvent(event: AgendaItem.Event): AuthResult<Unit> {
+        if (db.getEventDao().getEventById(event.id) == null ||
+            db.getEventDao().getModifiedEventById(event.id)?.modifiedType == ModifiedType.CREATE
+        ) {
+            db.getEventDao().upsertEvent(event.toEventEntity())
+            val result = getAuthResult {
+                api.createEvent(
+                    eventData = MultipartBody.Part
                         .createFormData(
-                            name = "photo$index",
-                            filename = eventPhoto.key,
-                            body = compressedImageByte.toRequestBody()
+                            name = "create_event_request",
+                            value = Json.encodeToString(event.toCreateEventRequest())
+                        ),
+                    photoData = event.photos.filterIsInstance<EventPhoto.Local>().mapIndexed {
+                        index, eventPhoto ->
+                        val uriByteConverter = UriByteConverter(
+                            appContext = appContext,
+                            ioDispatcher = ioDispatcher
+                        )
+                        val imageByte = uriByteConverter.uriToByteArray(
+                            uri = Uri.parse(eventPhoto.uri),
+                        )
+
+                        val compressedImageByte = BitmapCompressor().compressByteArray(
```

Comment:
If your class does not need constructor parameters, you can also make it an `object` 👍